### DISCUSSION TITLE
3D Plot of Elastic Modulus

### DISCUSSION THREAD

**Discussion by Savillian** [mtex-toolbox/mtex#996] [Ask Anything]
Hi everyone,

I have a large quantity of EBSD data that I would like to create a 3D spherical plot of the elastic modulus from. The aim of this is to show how the elastic modulus differs with orientation in the **specimen reference frame** and demonstrate anisotropy of this and other properties more easily.

I've already identified how to plot the elastic modulus on EBSD maps courtesy of another forum post, but I've been stumped on how to plot the effective elastic modulus (what would the material actually exhibit when pulled in tension along a given axis) on a spherical plot and/or stereographic projection given the orientation data from EBSD? Notionally I would like to plot this value for all possible orientations, or at least a refined grid of orientations.

I've included below some example code I've been working on, and where each attempt fails to achieve what I'm working towards.

Any help is appreciated, and thank you again for creating such a useful plug in!

Thanks in advance,
Alec

Attempt 1:
C = stiffnessTensor.load(fname2,ebsd('Titanium (Alpha)').CS)

% derive the crystal direction parallel to your reference direction
yCBD = inv(ebsd('Titanium (Alpha)').orientations).*yvector;

% compute the youngsmodulus parallel to those crystal directions
yMBD = YoungsModulus(C,yCBD);

% Plot Elastic Modulus on Orientations
figure
plot(yCBD,yMBD,'complete');
mtexColorbar('location','eastOutSide','title','Elastic Modulus (GPa)') **--> This returns a stereographic projection with crystal orientations and overlays the elastic modulus from the imported file. It does not however calculate an effective elastic modulus for a tensile load on this and other directions like I am looking for.**

Attempt 2:
% Identify inverses of each orientation
yA = inv(ebsd('Titanium (Alpha)').orientations); **--> This returns an error stating "Number of elements must not change."**

% Calculate the effective Young's Modulus for every orientation
Actual = YoungsModulus(C,yA);

% Plot the effective elastic modulus for each orientation in a stereographic projection
plot(Actual)
mtexColorbar('location','eastOutSide','title','Elastic Modulus (GPa)')

Attempt 3:
% Find solutions
[CVoigt, CReuss, CHill] = calcTensor(ebsd('Titanium (Alpha)'),C);
CHill = calcTensor(ebsd('Titanium (Alpha)'),C,'Hill');
CVoigt = calcTensor(ebsd('Titanium (Alpha)'),C,'Voigt');
CReuss = calcTensor(ebsd('Titanium (Alpha)'),C,'Reuss');

% Plot solution
figure
plot(CHill) % Returns pole figure like plot with wrong values **--> This returns a stereographic projection akin to what I expect, but the elastic modulus values are 40-50 GPA higher than they should be.**
mtexColorbar('location','eastOutSide','title','Elastic Modulus (GPa)') 

Attempt 4:
% Find solutions
[CVoigt2, CReuss2, CHill2] = calcTensor(odf_rotated,C);

% Plot solution
plot(CHill2) % Returns IPF like plot **--> This returns an inverse pole figure and not a stereographic projection**

**Comment by lfgmorales** [comment]
Hi

Maybe the command surf is what you are looking for

Have a look here

https://mtex-toolbox.github.io/PiezoElectricity.html

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 28 Apr 2021, at 18:25, AlphaTwister ***@***.***> wrote:

﻿

Hi everyone,

I have a large quantity of EBSD data that I would like to create a 3D spherical plot of the elastic modulus from. The aim of this is to show how the elastic modulus differs with orientation in the specimen reference frame and demonstrate anisotropy of this and other properties more easily.

I've already identified how to plot the elastic modulus on EBSD maps courtesy of another forum post, but I've been stumped on how to plot the effective elastic modulus (what would the material actually exhibit when pulled in tension along a given axis) on a spherical plot and/or stereographic projection given the orientation data from EBSD? Notionally I would like to plot this value for all possible orientations, or at least a refined grid of orientations.

I've included below some example code I've been working on, and where each attempt fails to achieve what I'm working towards.

Any help is appreciated, and thank you again for creating such a useful plug in!

Thanks in advance,
Alec

Attempt 1:
C = stiffnessTensor.load(fname2,ebsd('Titanium (Alpha)').CS)

% derive the crystal direction parallel to your reference direction
yCBD = inv(ebsd('Titanium (Alpha)').orientations).*yvector;

% compute the youngsmodulus parallel to those crystal directions
yMBD = YoungsModulus(C,yCBD);

% Plot Elastic Modulus on Orientations
figure
plot(yCBD,yMBD,'complete');
mtexColorbar('location','eastOutSide','title','Elastic Modulus (GPa)') --> This returns a stereographic projection with crystal orientations and overlays the elastic modulus from the imported file. It does not however calculate an effective elastic modulus for a tensile load on this and other directions like I am looking for.

Attempt 2:
% Identify inverses of each orientation
yA = inv(ebsd('Titanium (Alpha)').orientations); --> This returns an error stating "Number of elements must not change."

% Calculate the effective Young's Modulus for every orientation
Actual = YoungsModulus(C,yA);

% Plot the effective elastic modulus for each orientation in a stereographic projection
plot(Actual)
mtexColorbar('location','eastOutSide','title','Elastic Modulus (GPa)')

Attempt 3:
% Find solutions
[CVoigt, CReuss, CHill] = calcTensor(ebsd('Titanium (Alpha)'),C);
CHill = calcTensor(ebsd('Titanium (Alpha)'),C,'Hill');
CVoigt = calcTensor(ebsd('Titanium (Alpha)'),C,'Voigt');
CReuss = calcTensor(ebsd('Titanium (Alpha)'),C,'Reuss');

% Plot solution
figure
plot(CHill) % Returns pole figure like plot with wrong values --> This returns a stereographic projection akin to what I expect, but the elastic modulus values are 40-50 GPA higher than they should be.
mtexColorbar('location','eastOutSide','title','Elastic Modulus (GPa)')

Attempt 4:
% Find solutions
[CVoigt2, CReuss2, CHill2] = calcTensor(odf_rotated,C);

% Plot solution
plot(CHill2) % Returns IPF like plot --> This returns an inverse pole figure and not a stereographic projection

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/996>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN5MLCFT44XGADGHRV3TLAZF5ANCNFSM43XNICGQ>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/996", "url": "https://github.com/mtex-toolbox/mtex/discussions/996", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Savillian** [reply]
Hi Luiz,

Thanks for the suggestion! This creates a plot similar to that which I can already create with plot3d (see below), but maybe gives me more versatility in what I plot... I'll fiddle with this for a little while.
![Random0-4mmSimulationRemoved10AlphaTiModulus3D](https://user-images.githubusercontent.com/38051659/116462069-48e31f80-a826-11eb-954f-f7d2ba9cfac2.png)


The main challenge I'm having is how to apply orientation and intensity data simultaneously to these 3D plots, and make it so that my script automatically calculates the effective elastic modulus for any given set of rotations from the specimen reference frame. With the above plot I have the elastic modulus plotted, but this is only for a tensile load applied at the north and south poles and not accounting for orientation.

I know the issue lies somewhere in the syntax or calculation of what I'm selecting for one of the plotting attempts I listed out in the initial message, but I'm at a loss for what that may be (hence my inquiry). The Tensor Averages page has some similar data processing, but doesn't work towards applying the data in the same way I'm looking to.

Cheers,
Alec
